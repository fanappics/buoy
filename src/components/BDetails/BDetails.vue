<template>
<details>
  <summary>
    <span class="title">{{summary}}</span>
    <div class="icon">
      <slot name="icon"></slot>
    </div>
  </summary>
  <slot></slot>
</details>
</template>

<script>
export default {
  name: 'b-details',
  props: {
    summary: {
      type: String,
      required: true
    }
  }
}
</script>

<style>
.icon {
  float: right;
}
.title {
  font-weight: bold;
}
details[open] summary ~ * {
  animation: fadein 1s;
}
@keyframes fadein {
  from { opacity: 0; }
  to { opacity: 1; }
}
</style>

<docs>
```js
<b-details summary="The summary prop provides summary/title text.">
  <span slot="icon">&#10003;</span>
  <p>You can put your content into the default slot, and an optional icon into the named "icon" slot</p>
</b-details>
```
</docs>
