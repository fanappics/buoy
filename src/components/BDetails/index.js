import BDetails from './BDetails.vue'

/* istanbul ignore next */
BDetails.install = function install (Vue) {
  Vue.component(BDetails.name, BDetails)
}

export default BDetails
