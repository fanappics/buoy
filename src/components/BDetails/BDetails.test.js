import { mount } from 'vue-test-utils'
import BDetails from './BDetails'

describe('BDetails', () => {
  it('renders correctly', () => {
    const summary = 'hello world'
    const wrapper = mount(BDetails, {
      propsData: { summary }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
