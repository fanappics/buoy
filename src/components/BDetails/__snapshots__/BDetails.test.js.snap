// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BDetails renders correctly 1`] = `
<details>
    <summary><strong>hello world</strong>
        <div class="icon"></div>
    </summary>
</details>
`;
